dnstable_convert.c

```diff
@@ -1031,16 +1045,40 @@ main(int argc, char **argv)
 	long mmb = 0;
 	mtbl_compression_type compression = MTBL_COMPRESSION_ZLIB;
 	int compression_level = DEFAULT_COMPRESSION_LEVEL;
+	int dns_block_size = DNS_MTBL_BLOCK_SIZE;
+	int dnssec_block_size = DNSSEC_MTBL_BLOCK_SIZE;
+	int thread_count = 0;
+	struct mtbl_threadpool *pool = NULL;
 	const char *name = argv[0];
 	int c;
 
 	setlocale(LC_ALL, "");
 
-	while ((c = getopt(argc, argv, "Dc:l:m:prSs:")) != -1) {
+	while ((c = getopt(argc, argv, "b:Dc:l:m:t:prSs:")) != -1) {
 		mtbl_res res;
 		char *end;
 
 		switch(c) {
+		case 'b':
+			if (sscanf(optarg, "%d,%d", &dns_block_size, &dnssec_block_size) != 2 &&
+			    strchr(optarg, ',') != NULL) {
```

Comment:
This will still silently ignore an invalid number without a ','.  The full `sscanf` treatmet would be:
```
if (strchr(optarg, ',') != NULL) {
     if (sscanf(optarg, "%d,%d", ...) != 2) {
           /* error */
     }
} else if (sscanf(optarg, "%d", ...) != 1) {
     /* error */
}
```

I would suggest returning to `strtol`, like
```
dns_block_size = strtol(optarg, &endp, 10);
switch(*endp) {
default: 
     /* malformed block size */
     return (EXIT_FAILURE);
case '\0': break;
case ',': 
      optarg = endp;
      dnssec_block_size = strtol(optarg, &endp, 10);
      if (endp != '\0') {
           /*  malformed dnssec block size */
           return (EXIT_FAILURE);
     }
}
```

Reply:
I am confused here.  When I run this with an invalid block size (e.g. -1, 0, -22), the function does indeed return `EXIT_FAILURE`.  An invalid number for either block size is anything less than 1, which are the if statements that is immediately follow the initial `sscanf()`.

Reply:
You are correct when the input parses as a number, but is out of range (i.e., zero or negative).  If the `-b` argument doesn't parse as a number, `sscanf` returns 0, and the current logic will only reject the argument if it contains a comma.

Reply:
Revisited the `strtoul()` approach, works great now!